src/service/ampdoc-impl.js

```diff
@@ -756,6 +780,49 @@ export class AmpDocSingle extends AmpDoc {
   whenReady() {
     return this.readyPromise_;
   }
+
+  /** @override */
+  getMetaByName(name) {
+    if (!name) {
+      return null;
+    }
+
+    // 4.2.5.1 Standard metadata names
+    //   Names are case-insensitive, and must be compared in an ASCII case-
+    //   insensitive manner.
+    // https://html.spec.whatwg.org/multipage/semantics.html#standard-metadata-names
+    //
+    // Rules about case sensitivity and multiple meta tags with the same name
+    // are not as strict for non-standard metadata names. Adopt the conventions
+    // used by PHP get_meta_tags():
+    //   1. Special characters in the value of the name property are substituted
+    //      with '_', the rest is converted to lower case.
+    //   2. If two meta tags have the same name, only the last one is returned.
+    // https://www.php.net/manual/en/function.get-meta-tags.php
+    //
+    // Proper unicode support if browser can handle it
+    // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/unicode
+    const lowerName = name.toLowerCase();
+    const metaEls = toArray(
+      dev()
+        .assertElement(this.win.document.head)
+        .querySelectorAll(`meta[name]`)
+    );
+    const metaElIndex = findIndex(metaEls, el => {
+      const mName = el.getAttribute('name');
+      let sanitizedName;
+      try {
+        sanitizedName = mName.replace(/[^\x00-\x7F]/gu, '_').toLowerCase();
```

Comment:
This is overkill. If a publisher puts Unicode characters in the value of `name`, then unexpected behavior is reasonable (like `__` instead of `_`). Can just do `sanitizedName = mName.replace(/[^\x00-\x7F]/g, '_').toLowerCase();` in all browsers.

Reply:
Or maybe any sanitization at all is overkill. Thoughts?

Reply:
Woo I appreciate the thoroughness! 

Yea I think it's a bit overkill since this isn't a public API and there's a short list of valid `meta[name]` values (all of which are just ASCII). Probably better to save the bytes and CPU for users e.g. just use `meta[name="..."]` query selector.